plugin/src/main/java/org/opensearch/ml/action/upload_chunk/MLModelChunkUploader.java

```diff
@@ -0,0 +1,163 @@
+                                        .modelContentSizeInBytes(existingModel.getModelContentSizeInBytes())
+                                        .createdTime(existingModel.getCreatedTime())
+                                        .build();
+                                    IndexRequest indexRequest1 = new IndexRequest(ML_MODEL_INDEX);
+                                    indexRequest1.id(modelId);
+                                    indexRequest1
+                                        .source(
+                                            mlModelMeta
+                                                .toXContent(XContentBuilder.builder(XContentType.JSON.xContent()), ToXContent.EMPTY_PARAMS)
+                                        );
+                                    indexRequest1.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+                                    client.index(indexRequest1, ActionListener.wrap(re -> {
+                                        log.debug("Index model successfully {}", existingModel.getName());
+                                        semaphore.release();
+                                    }, e -> {
+                                        log.error("Failed to update model state", e);
+                                        semaphore.release();
+                                        listener.onFailure(e);
+                                    }));
+                                }
+                                listener.onResponse(new MLUploadModelChunkResponse("Uploaded"));
+                            }, e -> {
+                                log.error("Failed to upload chunk model", e);
+                                listener.onFailure(e);
+                            }));
+                        }, ex -> {
+                            log.error("Failed to init model index", ex);
+                            listener.onFailure(ex);
+                        }));
+                    } catch (Exception e) {
+                        log.error("Failed to parse ml model" + r.getId(), e);
+                        listener.onFailure(e);
+                    }
+                } else {
+                    listener.onFailure(new MLResourceNotFoundException("Failed to find model"));
+                }
+            }, e -> {
+                if (e instanceof IndexNotFoundException) {
+                    listener.onFailure(new MLResourceNotFoundException("Failed to find model"));
+                } else {
+                    log.error("Failed to get ML model " + modelId, e);
+                    listener.onFailure(e);
+                }
+            }));
+        } catch (Exception e) {
+            log.error(e.getMessage());
+            listener.onFailure(e);
+        }
+    }
+
+    public static String customModelId(String modelId, Integer chunkNumber) {
+        return modelId + "_" + chunkNumber;
+    }
+
+    private boolean isChunkExceeding10MB(final long length) {
```

Comment:
We may change the size limit in future. How about changing to a more generic name like "validateChunkSize"

Reply:
Sure Yaliang. Makes more sense.